src/workflow/dtos/add-node.input.ts

```diff
@@ -17,6 +20,7 @@ export class AddNodeInputPipe implements PipeTransform<AddNodeInput, Promise<Add
 
   async transform(value: AddNodeInput): Promise<AddNodeInputFull> {
     const service = await this.dampLabServicePipe.transform(value.serviceId);
-    return { ...value, service, state: WorkflowNodeState.QUEUED };
+    const price = value.price ?? service.price;
```

Comment:
I see what you were up to here, but actually I think that if the `value.price` is undefined - meaning that when the client submitted their job, they weren't seeing a price for that particular service - then we should leave it undefined in the snapshot; probably the service price was updated after the job was submitted!